nodeEditor/src/graphEditor.tsx

```diff
@@ -475,6 +495,47 @@ export class GraphEditor extends React.Component<IGraphEditorProps, IGraphEditor
         });
     }
 
+    addFrameFromSerialization(editorData: IEditorData , source: any, rootUrl: string = "") {
+        let map: {[key: number]: NodeMaterialBlock} = {};
+
+        // Create blocks
+        for (var parsedBlock of source.blocks) {
+            let blockType = _TypeStore.GetClass(parsedBlock.customType);
+            if (blockType) {
+                let block: NodeMaterialBlock = new blockType();
+                block._deserialize(parsedBlock, this.props.globalState.nodeMaterial.getScene(), rootUrl);
+                map[parsedBlock.id] = block;
+
+                this.createNodeFromObject(block);
+            }
+        }
+
+        if (source.locations || source.editorData && source.editorData.locations) {

```

Comment:
I feel like we should not duplicate that code but instead reuse the one we already have
We could update loadFromSerialization to have a flag that let the function knows we do not want to clear the current graph (so skip the this.clear) and instead add loaded data to current data

My ask was to not mention anything frame related in bjs core as the concept of frame does not exist there but it is ok to update the existing functions to make them more flexible